saleor/graphql/order/mutations/order_grant_refund_create.py

```diff
@@ -182,6 +187,32 @@ def validate_input(cls, input: dict[str, Any]):
                 }
             )
 
+    @classmethod
+    def _resolve_refund_reason_instance(
+        cls, /, reason_reference_id: str, refund_reason_reference_type: str
+    ):
+        reason_reference_pk = cls.get_global_id_or_error(
+            reason_reference_id, only_type="Page", field="reason_reference"
+        )
+
+        try:
+            reason_reference_instance = Page.objects.get(
+                pk=reason_reference_pk,
+                page_type=refund_reason_reference_type,  # type: ignore[misc]
```

Comment:
`# type: ignore[misc]` hides the issue that you have here:
`page_type` probably accept `PageType` object.  
`refund_reason_reference_type` is typed as `str`
in line 308, you call `refund_reason_reference_type.pk`, and it returns `int` 

The input should accept `page_type_id` (`int`) or `page_type` (`PageType`).
Django `.get` should recieve:
`.get(page_type_id=page_type_id)` OR `.get(page_type=page_type)`

In both cases, `type: ignore` should be removed :)